<launch>
    <arg name="own_name" default="car30" />
    
    <!-- Total number of cars (including self) -->
    <arg name="car_count" default = "4"/>
    
    <!-- Other car names (if car_count is 2, then nodes will read own_name, other_car1, then disregard the rest) -->
    <arg name="other_car1" default = "car38"/>
    <arg name="other_car2" default = "car39"/>
    <arg name="other_car3" default = "car34"/>
    <arg name="other_car4" default = "car5"/>
    <arg name="other_car5" default = "car6"/>
    <arg name="other_car6" default = "car7"/>
    <arg name="other_car7" default = "car8"/>

    <arg name="config_file" default="$(find nhttc_ros)/config/nhttc_real.yaml" />
    <rosparam file="$(arg config_file)" command="load" />

    <!-- Set to 1 if you want to run the map_server -->
    <arg name="map_server" value = "0"/>

    <!-- Launch  map server-->
    <group if="$(arg map_server)">
        <include file="$(find mushr_sim)/launch/map_server.launch" />
    </group>

    <!-- Launch Teleop -->
    <group>
	<include file="$(find mushr_base)/launch/teleop.launch">
	        <arg name="car_name" value="$(arg own_name)" />
	</include>
    </group>

    <!-- Launch Own Controller -->
    <group ns="$(arg own_name)">
        <node pkg="nhttc_ros" type="nhttc_ros_node" name="controller" output="screen">
            <param name="max_agents" value="$(arg car_count)"/>
            <param name="car_name" value="$(arg own_name)"/>
            <param name="sim" value="false"/>
        </node>
    </group>

    <!-- <node pkg="nhttc_ros" type="nhttc_pose_goal_init.py" name="initializer" output="screen">
        <param name="car_count" value="$(arg car_count)"/>
        Each arg is a tuple of (car_name,boolean) of whether it is a sim car or not
        <param name="car1_name-sim" value="$(arg own_name),0"/>
        <param name="car2_name-sim" value="$(arg other_car1),0"/>
        <param name="car3_name-sim" value="$(arg other_car2),0"/>
        <param name="car4_name-sim" value="$(arg other_car3),0"/>
        <param name="car5_name-sim" value="$(arg other_car4),1"/>
        <param name="car6_name-sim" value="$(arg other_car5),1"/>
        <param name="car7_name-sim" value="$(arg other_car6),1"/>
	<param name="car8_name-sim" value="$(arg other_car7),1"/>
    </node> -->
    <!-- <node pkg="nhttc_ros" type="nhttc_publisher.py" name="waypointer"/> -->

</launch>
